code/go/0chain.net/smartcontract/interestpoolsc/node.go

```diff
@@ -100,11 +100,38 @@ func (gn *GlobalNode) set(key string, value string) error {
 		}
 		gn.MaxMint = state.Balance(fValue * 1e10)
 	default:
+		if strings.HasPrefix(key, Settings[Cost]) {
```

Comment:
do prefix check in setCostValue method

Reply:
That will not work as I need the return statement to work only if it has the prefix, otherwise I need it to return the error from the next line.